util/yaml/yaml.go

```diff
@@ -32,6 +33,27 @@ import (
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha2"
 )
 
+func ExtractClusterReferences(out *ParseOutput, c *clusterv1.Cluster) (res []*unstructured.Unstructured) {
+	if c.Spec.InfrastructureRef != nil {
```

Comment:
I prefer to avoid nesting as much as possible - I find this cleaner. WDYT?

```go
if c.Spec.InfrastructureRef == nil {
  return nil
}

obj := out.FindUnstructuredReference(c.Spec.InfrastructureRef)
if obj == nil {
  return nil
}

return []*unstructured.Unstructured{obj}
```

Reply:
That's fine